addons/l10n_uy/models/res_partner.py

```diff
@@ -0,0 +1,99 @@
+        # Catch use case where the record label is about the public user (name: False)
+        if 'False' not in partner_label:
+            msg = _(
+                '\nThe %(vat_label)s number [%(wrong_vat)s] for %(record_label)s does not seem to be valid. \nNote:'
+                ' the expected format is %(expected_format)s',
+                vat_label=vat_label,
+                wrong_vat=partner.vat,
+                record_label=partner_label,
+                expected_format=expected_format,
+            )
+        else:
+            msg = _(
+                '\nThe %(vat_label)s number [%(wrong_vat)s] does not seem to be valid. \nNote: the expected format'
+                ' is %(expected_format)s',
+                vat_label=vat_label,
+                wrong_vat=partner.vat,
+                expected_format=expected_format,
+            )
+        return msg
+
+    def _l10n_uy_nie_ci_is_valid(self):
+        """ Check if a NIE or CI number is a valid one.
+
+        CI:
+            1) The ID number is taken up to the second to last position, that is, the first 6 or 7 digits.
+            2) Each digit is multiplied by a different factor starting from right to left, the factors are:
+                2, 9, 8, 7, 6, 3, 4.
+            3) The products obtained are added:
+            4) The base module 10 is calculated on this result to obtain the check digit, expressed in another way,
+            the next number ending in zero is taken that follows the result of the addition (for the example
+            would be 60) subtracting the sum itself: 60 - 59 = 1. The verification digit of the example ID is 1.
+
+            NOTE: If the ID has fewer digits, it is preceded with zeros and the mechanism described above is applied
+
+        NIE:
+            The calculation for the NIE is the same as that used for the CI. The only difference is that we skip the
+            first number
+
+        Both algorithms where extracted from Uruware's Technical Manual (section 9.2 and 9.3)
+
+        Return: False is not valid, True is valid
+        """
+        self.ensure_one()
+
+        # If we have vat but this one does not have numbers then is an invalid one
+        ci_nie_number = re.sub('[^0-9]', '', self.vat)
+        if not ci_nie_number:
+            return False
+
+        # we get the validation digit, if NIE doc type we skip the first digit
+        is_nie = self.l10n_latam_identification_type_id.l10n_uy_dgi_code == '1'
+        ci_nie_number, verif_digit = ci_nie_number[1 if is_nie else 0:-1], int(ci_nie_number[-1])
+
+        # If number is < 7 digits we add 0 to the left
+        ci_nie_number = "%07d" % int(ci_nie_number)
+
+        # If NIE > 7 digits is not valid
+        if len(ci_nie_number) > 7:
+            return False
+
+        verification_vector = [2, 9, 8, 7, 6, 3, 4]
+        num_sum = sum(int(ci_nie_number[index]) * verification_vector[index] for index in range(6))
```

Comment:
If we are only checking 6 numbers (from `range(6)`), why is there 7 numbers inside `verification_vector`? Or should it be `range(7)` instead?
```suggestion
        verification_vector = (2, 9, 8, 7, 6, 3, 4)
        num_sum = sum(int(ci_nie_number[i]) * verification_vector[i] for i in range(6))